tools/FakeItEasy.Deploy/Program.cs

```diff
@@ -0,0 +1,104 @@
+                ?? throw new Exception($"Can't find release {releaseName}");
+
+            var artifactsFolder = Path.GetFullPath(Path.Combine(GetCurrentScriptDirectory(), "../../artifacts/output/"));
+            var artifactsPattern = "*.nupkg";
+
+            var artifacts = Directory.GetFiles(artifactsFolder, artifactsPattern);
+            if (!artifacts.Any())
+            {
+                throw new Exception("Can't find any artifacts to publish");
+            }
+
+            Console.WriteLine($"Uploading artifacts to GitHub release {releaseName}");
+            foreach (var file in artifacts)
+            {
+                await UploadArtifactToGitHubReleaseAsync(gitHubClient, release, file);
+            }
+
+            Console.WriteLine("Pushing nupkgs to nuget.org");
+            foreach (var file in artifacts)
+            {
+                await UploadPackageToNuGetAsync(file);
+            }
+
+            Console.WriteLine("Finished deploying");
+        }
+
+        private static async Task UploadArtifactToGitHubReleaseAsync(GitHubClient client, Release release, string path)
+        {
+            var name = Path.GetFileName(path);
+            Console.WriteLine($"Uploading {name}");
+            using (var stream = File.OpenRead(path))
+            {
+                var upload = new ReleaseAssetUpload
+                {
+                    FileName = name,
+                    ContentType = "application/octet-stream",
+                    RawData = stream,
+                    Timeout = TimeSpan.FromSeconds(100)
+                };
+
+                var asset = await client.Repository.Release.UploadAsset(release, upload);
+                Console.WriteLine($"Uploaded {asset.Name}");
+            }
+        }
+
+        private static async Task UploadPackageToNuGetAsync(string path)
+        {
+            const string nugetServer = "https://www.nuget.org/api/v2/package";
+            string name = Path.GetFileName(path);
+            Console.WriteLine($"Pushing {name}");
+            await RunAsync(ToolPaths.NuGet, $"push \"{path}\" -ApiKey {GetNuGetApiKey()} -Source {nugetServer} -NonInteractive -ForceEnglishOutput", noEcho: true);
+            Console.WriteLine($"Pushed {name}");
+        }
+
+        private static (string repoOwner, string repoName) GetRepositoryName()
+        {
+            var repoNameWithOwner = Environment.GetEnvironmentVariable("APPVEYOR_REPO_NAME");
+            var parts = repoNameWithOwner.Split('/');
+            return (parts[0], parts[1]);
+        }
+
+        private static GitHubClient GetAuthenticatedGitHubClient()
+        {
+            var token = GetGitHubToken();
+            var credentials = new Credentials(token);
+            return new GitHubClient(new ProductHeaderValue("FakeItEasy-build-scripts")) { Credentials = credentials };
+        }
+
+        private static string GetGitHubToken() => Environment.GetEnvironmentVariable("GITHUB_TOKEN");

```

Comment:
It annoys me that we have a completely different way to get the GitHub token than in PrepareRelease... but the `.githubtoken` file isn't available on AppVeyor, and the environment variable usually isn't set locally, so I guess it makes sense.

Reply:
Yup. I'd set the environment variable at home if you prefer that workflow, though.

Reply:
Lovely.